### ISSUE TITLE
Most of boost can be swapped for C++11/14

### ISSUE THREAD

**Issue by ensiform** [jswigart/omni-bot#7]
Most of the boost usage can be removed in favor or C++11 / C++14.  MSVC 2013 and 2015 should support everything needed.  FS obviously needs to stay (IDK about ENABLE_DOWLOAD stuff) and/or interprocess if that is still being used, dynamic_bitset seems to be the only thing not replaceable.

Probably don't need for interface code at all.

**Comment by plastovicka** [comment]
Most dedicated game servers run on Linux. We have already tried to use newer version of gcc. Some administrators immediately responded that they could not start the game because of missing GLIBC or LIBSTDC++. We had to revert and return to old version of gcc.
RTCW and Enemy Territory are very old games. Some servers are running for years and admins do not want to update system libraries.

**Comment by ensiform** [comment]
Hmm, I see?  We're forcing the hand with C++11 for OpenJK now which is for Jedi Academy.  Just how old of software are these servers running...  Not to mention, why?  Are they servers that see lots of players to matter too?

**Comment by plastovicka** [comment]
Do you really want to use Omni-bot in Jedi Academy ? I think it would be very hard to convert Omni-bot interface and goal scripts.

**Comment by jswigart** [comment]
We had Omnibot working in Jedi Academy years ago and using some of the
force powers. But I think he's talking about the mod itself. Linux has the
issue that it is more invasive to update the libraries than it is in
Windows. For Windows you just have to install the 2013 or whatever
redistributables. For whatever reason it's not as easy I guess in Linux.

The advanced stuff I'm doing in the main bot branch is using some more of
modern c++. I'm using thread, regex, and file system stuff that's not in
std. I've already converted to unordered map and a few other things.

**Comment by plastovicka** [comment]
I didn't know that Omni-bot is working in Jedi Academy. Could you write some info on the wiki ?
http://omni-bot.invisionzone.com/wiki/index.php?title=Supported_Games

**Comment by jswigart** [comment]
We never released anything yet so it didn't seem right to advertise it.

**Comment by ensiform** [comment]
Regex actually is in C++11, but I tried swapping out std::hash_map for std::unordered_map and it fails.

Also, Jedi Academy using newer C++ doesn't change things for OB.  I was just trying to avoid using boost with VS 2015.

JA support was only for my old mod at one time.  It would need to be re-added to an OpenJK fork at some point.

I got the 0.83 JA dll compiling under 2015 now with boost (1.59.0).  But it gives fail to load FS error.  It found a lot of variable shadowing.

**Comment by Dragonji** [comment]
> Most dedicated game servers run on Linux. We have already tried to use newer version of gcc. Some administrators immediately responded that they could not start the game because of missing GLIBC or LIBSTDC++. We had to revert and return to old version of gcc.
> RTCW and Enemy Territory are very old games. Some servers are running for years and admins do not want to update system libraries.

Come on, even all the time outdated Debian (if we talk about packages) has gcc 4.9 already: https://packages.debian.org/pl/stable/gcc

There's no point in "supporting" unsecure servers that use ancient OSes.

**Comment by jswigart** [comment]
I agree to large extent. There's only so much I'm willing to be held back by the limitations of particular deployment environments. I'm not sure why static linking isn't more of a solution to these dependencies on Linux as it is with Windows.

**Comment by ensiform** [comment]
Because security is higher priority than static linkage.  And static linkage does not allow for security patches without manually updating every package that is used by your product and recompiling them + your product.

**Comment by jswigart** [comment]
That's great in theory. But as we see the added pita of all the interdependencies ends up meaning lots of users don't bother to update at all. I see this same issue in commercial environments. It's an OS enforced "dll hell" that most people don't bother keeping up with.

**Comment by Gustavo6046** [comment]
A summary of the stakes at hand here; Boost is an extra dependency, but including it means maintaining compatibility with less recent compilers (of the likes of those used in most id Tech 3-related software back in the day).
